torch/csrc/jit/stack.h

```diff
@@ -42,21 +42,37 @@ static inline IValue pop(Stack & stack) {
   return r;
 }
 
+// variadic pop:
+// int64_t a; at::Tensor b;
+// pop(stack, a, b);
+// equivalent to:
+// b = pop(stack).toTensor();
+// a = pop(stack).toInt();
+template<typename... Types>
+static inline void pop(Stack& stack, Types&... args) {
+  size_t i = 0;
+  constexpr size_t N = sizeof...(args);
+  int result[N] = {
+    (args = std::move(peek(stack,i++, N)).template to<Types>(),0)...
+  };
+  (void) result;
+  drop(stack, N);
+}
+template<typename... Types>
+static inline void push(Stack& stack, Types... args) {
+  constexpr size_t N = sizeof...(args);
+  int result[N] = {
+    (stack.push_back(std::forward<Types>(args)), 0)...
+  };
```

Comment:
This is what the official C++ docs recommended as the pattern. Initializer lists, _but not arguments to functions_, have sequence points between their commas. This is important because each of those push_backs is side-effectful.

Reply:
Not sure about the evaluation order of the comma operator either, so Ill stick with what was recommended.

Reply:
I'm pretty sure the comma enfores evaluation order, but let's do the recommended thing.